src/main/scala/chisel3/util/Decoupled.scala

```diff
@@ -239,6 +244,17 @@ class Queue[T <: Data](val gen: T,
     maybe_full := do_enq
   }
 
+  if (hasFlush) { //flushable queue specific logic
+    when(io.flush.get) {
+      enq_ptr.reset()
+      deq_ptr.reset()
+    }
+
+    when(io.flush.get) {
+      maybe_full := false.B
+    }
+  }
+
   io.deq.valid := !empty
   io.enq.ready := !full
```

Comment:
what should behavior of `ready`, and `valid`, be relative to the flush? As @sequencer noted above, can you flush at the same time that you actually enqueue something (seems like the answer shoudl be "no"). I think it's ok to `valid` something the same time though

Reply:
Yeah Deborah and I looked at it and we agree, and I'm working on making a test case and updating the logic now!

Reply:
If don't support this, I think `assert(enq && maybe_full)` is enough.
If support, just make `enq` and `deq` in the `otherwise` block of `flush` is OK.(but it will increase the critical path. with a `nand` gate.)

Reply:
i think we can resolve this comment